---
title: "SCDC: Bulk RNA-seq deconvolution by scRNA-seq using an ENSEMBLE method"
author: "Meichen Dong"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{SCDC: Bulk RNA-seq deconvolution by scRNA-seq}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Single cell demographic and clustering QC
For each singl-cell dataset with raw counts, we can first explore the demographic information by the visualization function `DemoPlot`. Here, single cells from each subject are summarized by total counts, counts per cell type, and the overall composition. This gives us the general understanding or a dataset.
Use single cell dataset Segerstolpe et al. as an example, we can visualize the number of single cells and the relative cell-type proportion per subject.
```{r, fig.height = 4, fig.width = 5, fig.align = "center"}
setwd("H:/994/SCDC/data")
library(SCDC)
library(ggplot2)
library(pheatmap)
library(L1pack) # lad
library(reshape2) # melt
library(nnls)

cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
ct1 <- c("mediumorchid1","mediumpurple1","lightskyblue","seagreen1","yellow","tan1","azure3")
seger <- readRDS("S0102_SEGER.rds")
DemoPlot(seger, cluster = "cluster", sample = "sample", select.ct = c("alpha","beta","delta","gamma","ductal","acinar"), Palette = ct1)
```

Also, we can perform the clustering QC to filter out single cells that are potential doublets, miss-classified, or belonging to cell types not of interest.
The threshold of keeping a single cell from an assigned cluster is chosen as 0.7 in this illustration example. This user-defined threshold should be selected carefully, to achieve the effect of quality control of single cells without obstructing the performance on the downstream analysis. An unproper threshold could potentially lead to filtering out a large proportion of cells from rare cell types. To be conservative, this can be chosen from (0.5,0.7).
```{r, message=F, fig.height = 4, fig.width = 4, fig.align = "center"}
seger.qc <- my_qc(seger, ct.varname = "cluster", sample = "sample", scsetname = "Segerstolpe", ct.sub = c("alpha","beta","delta","gamma","ductal","acinar"), qcthreshold = 0.7)
```
Similarly we can perform clustering QC for single cells from only one subject using function `SCDC_qc_ONE`.


## Data Input
We will use the single cell dataset after clustering-QC.
Pancreas data: 

-- Single cells (only selected healthy subjects): Baron et al., Segerstolpe et al., Xin et al.

-- Real bulk samples: Segerstolpe et al., Fadista et al.

Mammary gland data:

-- Single cells (only selected healthy subjects): Perou's lab, Tabula Muris

-- Real bulk samples: Perou's lab

Three-cell-line mixture. Single cells and real bulk sample are cultured by ratio of 6:3:1 for MDA-MB468: MCF7: normal fibroblast cells.

```{r}
## Pancreas data
## Single cells:
setwd("H:/994/SCDC/data")
qc.baron<- readRDS( "S0103_QCBARON.rds") #after CQC
qc.seger<- readRDS( "S0103_QCSEGER.rds") #after CQC
qc.xin<- readRDS( "S0103_QCXIN.rds") #after CQC

## real bulk samples
fadista_77 <- readRDS("H:/994/papers/baron/data/fadista_77.rds")

## Mammary Gland data
## Single cells
qc.perou<- readRDS("S0103_QCPEROU.rds") #after CQC
qc.tmouse<- readRDS("S0103_QCTMOUSE.rds") #after CQC
## real bulk samples
bulk10x <- readRDS("S0102_peroubulk10x_fvb34.rds")
bulkff <- readRDS("S0102_peroubulkff_fvb34.rds")


## Three-cell-line mixture
## Single cells
qc.3cl<- readRDS("S0103_QC3CL.rds") #after CQC
sc.3cl <- readRDS("H:/994/papers/perou/data/P0113_3clESET.rds")
## real bulk sample
bulk3cl <- readRDS("H:/994/papers/perou/data/P0113_bulkESET.rds")
```


## Simulation
SCDC allows users to construct pseudo bulk samples in two ways:

(1) Sum up gene-wise raw read counts per subject, and the true cell-type proportion is consistent with the single cell clustering result.

(2) Randomly sample single cells from each of the cell types of interest without replacement, and then sum up raw read counts per subject. The cell-type proportions are different from single cell clustering result.
```{r, message=F}
## Use cluster-QCed single cell dataset for downstream analysis
## (1) using all single cells
pseudo.seger <- generateBulk_allcells(qc.seger$sc.eset.qc, ct.varname = "cluster", sample = "sample", ct.sub = c("alpha","beta","delta","gamma","ductal","acinar"))

## (2) using random proportions of single cells
pseudo.seger.rand <- generateBulk_norep(qc.seger$sc.eset.qc, ct.varname = "cluster", sample = "sample", ct.sub = c("alpha","beta","delta","gamma","ductal","acinar"), nbulk = 3)

## we know the true relative proportions for the pseudo bulk samples:
round(pseudo.seger$truep,3)
round(pseudo.seger.rand$true_p,3)
```

## Deconvolution and ENSEMBLE
We will use three real bulk sample datasets to illustrate SCDC and ENSEMBLE method.

###Pancreas data

(1) Deconvolution of pseudo bulk samples using different reference datasets.

(2) ENSEMBLE of pseudo bulk samples

(3) Deconvolution of real bulk samples: Fadista et al.

```{r, message=F}
pseudo.seger <- generateBulk_allcells(qc.seger$sc.eset.qc, ct.varname = "cluster", sample="sample", ct.sub = c("alpha","beta","delta","gamma"))

### (1)
bulkseger.scseger <- SCDC_prop(bulk.eset = pseudo.seger$pseudo_eset, sc.eset = qc.seger$sc.eset.qc, ct.varname = "cluster", sample = "sample", ct.sub = c("alpha","beta","delta","gamma"), truep = pseudo.seger$truep)
bulkseger.scbaron <- SCDC_prop(bulk.eset = pseudo.seger$pseudo_eset, sc.eset = qc.baron$sc.eset.qc, ct.varname = "cluster", sample = "sample", ct.sub = c("alpha","beta","delta","gamma"), truep = pseudo.seger$truep)
bulkseger.scxin <- SCDC_prop(bulk.eset = pseudo.seger$pseudo_eset, sc.eset = qc.xin$sc.eset.qc, ct.varname = "cluster", sample = "sample", ct.sub = c("alpha","beta","delta","gamma"), truep = pseudo.seger$truep)

bulkseger.scseger$peval$evals.table
bulkseger.scbaron$peval$evals.table
bulkseger.scxin$peval$evals.table
```
We can observe the discrepancy of performance using different datasets. It's clear that using single cells from the reference dataset Segerstolpe et al., we derive the highest Pearson correlation, and the lowest RMSD and mAD.

Next, we will explore the ENSEMBLE results. Using the pseudo bulk samples constructed above, we try to deconvolve using three reference datasets: Baron et al., Segerstolpe et al. and Xin et al. The search length is a parameter that is related to the grid search method. We are evaluating the performance of each possible combination of the deconvolution results, and this search length decides how carefully we will search through the combinations. For instance, when search.length = 0.01, the weights for the three reference datasets could be (0.01,0,0.99).

```{r, message=F}
### (2)
pancreas.sc <- list(baron = qc.baron$sc.eset.qc,
                    seger = qc.seger$sc.eset.qc,
                    xin = qc.xin$sc.eset.qc)

PS_ensemble <- SCDC_ENSEMBLE(bulk.eset = pseudo.seger$pseudo_eset, sc.eset.list = pancreas.sc, ct.varname = "cluster",
                             sample = "sample", truep = pseudo.seger$truep[complete.cases(pseudo.seger$truep),], ct.sub =  c("alpha","beta","delta","gamma"), search.length = 0.01)
PS_ensemble$w_table
```
`SCDC_ENSEMBLE` provides the weights selected by the grid search method, as well as the mathematical solutions by LAD regression and the NNLS regression. We derived consistent weights for the three reference datasets. The reference dataset Segerstolpe et al. derives a weight of 1 from the LAD regression. This means that we tend to choose the reference dataset that is from the same source as the pseudo bulk samples.

For real bulk samples from Fadista et al., we analyze them by disease conditions in the ENSEMBLE step. It might take a while if you include the grid search method, hence, we directly show the results here. `w1` is the weight for single cell reference from Baron et al., and `w2` is for single cell reference from Segerstolpe et al. From previous simulation results, SCDC, as its default, chooses the LAD regression weights as the final weights. For example, when ensembling results for the healthy samples, we assign 0.477 to the reference dataset Baron et al., and 0.523 to the reference dataset Segerstolpe et al.

```{r, message=F}
# Fadista et al. real bulk samples
# baron.h <- qc.baron$sc.eset.qc
# seger.h <- qc.seger$sc.eset.qc
# F_ensemble_h2 <- SCDC_ENSEMBLE(bulk.eset = fadista_77[,fadista_77$hba1c_class2 == "Normal"], sc.eset.list = list(baronh = baron.h, segerh = seger.h), ct.varname = "cluster",
#                                sample = "sample", truep = NULL, ct.sub =  c("alpha","beta","delta","gamma","acinar","ductal"), search.length = 0.01)  
# 
# F_ensemble_d2 <- SCDC_ENSEMBLE(bulk.eset = fadista_77[,fadista_77$hba1c_class2 == "T2D"], sc.eset.list = list(baronh = baron.h, segerh = seger.h), ct.varname = "cluster",
#                                sample = "sample", truep = NULL, ct.sub =  c("alpha","beta","delta","gamma","acinar","ductal"), search.length = 0.01)  
F_ensemble_h2 <- readRDS("H:/994/SCDC/data/S0103_QC_fadistaens_healthy2.rds")
F_ensemble_d2 <- readRDS("H:/994/SCDC/data/S0103_QC_fadistaens_diabetic2.rds")
F_ensemble_d2$w_table
F_ensemble_h2$w_table
```

To visualize the ENSEMBLE result, we provide the boxplot as follows:
```{r, message=F, fig.height = 4, fig.width = 6, fig.align = "center"}
## create function that integrate the ENSEMBLE results:
getPropBox <- function(ens_h, ens_d, w1h, w2h, w1d,w2d, ref){
  prop.h <- ens_h$prop.list[[1]]$prop.est.mvw*w1h + ens_h$prop.list[[2]]$prop.est.mvw*w2h
  prop.h <- as.data.frame(prop.h)
  
  prop.d <- ens_d$prop.list[[1]]$prop.est.mvw*w1d + ens_d$prop.list[[2]]$prop.est.mvw*w2d
  prop.d <- as.data.frame(prop.d)
  
  prop2 <- rbind(prop.h, prop.d)
  prop2$condition <- c(rep("Normal", nrow(prop.h)), rep("T2D", nrow(prop.d)))
  
  dtmelt <- melt(prop2, id.vars = "condition")
  dtmelt$ref <- as.factor(ref)
  return(dtmelt)
}
fdt.ens <- getPropBox(ens_h = F_ensemble_h2, ens_d = F_ensemble_d2, w1h = 0.41, w2h = 0.59, w1d = 0.48, w2d = 0.52, ref = "ENSEMBLE")
fdt.seger <- getPropBox(ens_h = F_ensemble_h2, ens_d = F_ensemble_d2, w1h = 0, w2h =1, w1d = 0, w2d = 1, ref = "Segerstolpe")
fdt.baron <- getPropBox(ens_h = F_ensemble_h2, ens_d = F_ensemble_d2, w1h = 1, w2h =0, w1d = 1, w2d = 0, ref = "Baron")
dtbox_fa <- rbind(fdt.ens, fdt.seger, fdt.baron)

## boxplot:
ggplot(dtbox_fa, aes(x=variable, y=value, color = condition)) + 
  geom_boxplot() + geom_jitter(aes(x=variable, color = condition),position = position_dodge(0.8), alpha = 0.2)+
  theme(axis.text.x = element_text(angle = 30, hjust = 1, size=10),
        axis.text.y = element_text(size = 10),
        text = element_text(size = 10),
        plot.title = element_text(size=10, face = "bold"),
        plot.margin=unit(c(1,1,-5,0), "mm"),
        legend.position="top",legend.title = element_blank(),
        legend.text = element_text(size=8),
        legend.box.spacing = unit(0, "mm"))+
  ylab("") + xlab("") + facet_grid(rows = vars(ref))
```

Next, we explore of cell-type proportion and the HbA1c levels. In detail, for each of the cell types, we fit linear model with the estimated proportions as outcome and age, gender, BMI, HbA1c level as predictors.
```{r, message=F,fig.height = 5, fig.width = 8, fig.align = "center"}
getPropbycond <- function(ens_h, ens_d, w1h, w2h, w1d,w2d, ref, hba1clvl){
  prop.h <- ens_h$prop.list[[1]]$prop.est.mvw*w1h + ens_h$prop.list[[2]]$prop.est.mvw*w2h
  prop.h <- as.data.frame(prop.h)
  
  prop.d <- ens_d$prop.list[[1]]$prop.est.mvw*w1d + ens_d$prop.list[[2]]$prop.est.mvw*w2d
  prop.d <- as.data.frame(prop.d)
  
  prop2 <- rbind(prop.h, prop.d)
  prop2$condition <- c(rep("Normal", nrow(prop.h)), rep("T2D", nrow(prop.d)))
  return(prop2)
}
## combine demographic information
fadista_ens_prop <- getPropbycond(ens_h = F_ensemble_h2, ens_d = F_ensemble_d2, w1h = 0.41, w2h = 0.59, w1d = 0.48, w2d = 0.52, ref = "ENSEMBLE")
fadista_demo <- fadista_77@phenoData@data[,c("age","gender","hba1c","bmi")]
fadista_ens_all <- cbind(fadista_ens_prop, fadista_demo[rownames(fadista_ens_prop),])
getlmtable <- function(ct){
  fadista_ens_all$ct <- fadista_ens_all[,ct]
  lm.ens1 <- lm(ct ~ hba1c + age + bmi + gender, data = fadista_ens_all)
  s1 <- summary(lm.ens1)
  sdt <- cbind(round(s1$coefficients,4), celltype = rep(ct,5))
  return(sdt)
} 

dtlm <- NULL
for (ct in c("alpha","beta","delta","gamma","acinar","ductal")){
  dtlm0 <- getlmtable(ct)
  dtlm <- rbind(dtlm, dtlm0)
}

dat_text_fad <- data.frame(label = paste("p-value =",dtlm[seq(from = 2 , to = 27, by=5),c(4)]),
                           variable = dtlm[seq(from = 2 , to = 27, by=5),c(5)],
                           hba1c = 6, value = 0.5,
                           condition = "T2D")
dtmelt <- melt(fadista_ens_all, id.vars = c("condition","age","gender","bmi","hba1c"), measure.vars = c("alpha","beta","delta","gamma","acinar","ductal"))
ggplot(dtmelt[dtmelt$variable %in% c("alpha","beta","delta","gamma","acinar","ductal"),], aes(x=hba1c, y= value)) + geom_point(aes(color = condition)) + 
  geom_smooth(method='lm', se = FALSE, color = "black", lwd = 0.25) +
  theme(legend.position = "top", legend.title = element_blank(), axis.title.x = element_blank(),
        legend.box.spacing = unit(0, "mm"), axis.title.y = element_blank()) + 
  facet_grid(cols = vars(variable), scales = "free") +
  geom_text(data = dat_text_fad[dat_text_fad$variable %in% c("alpha","beta","delta","gamma","acinar","ductal"),], 
            mapping = aes(x=5.5, y=0.75,label = label),
            hjust   = 0.2, vjust   = 0)
```
We compared the ENSEMBLE results to the testing results using a uni-reference dataset. The listed p-values are for the HbA1c levels in the linear model where beta cell proportions are the outcome variable. The `ENSEMBLE` method and the uni-reference dataset Segerstolpe et al. would both lead to the conclusion that beta cell proportions are associated with the HbA1c levels. 
Moreover, the p-value in the ENSEMBLE beta cell proportion model is 0.0023<0.0054, meaning that this statistical evidence is even stronger than that given be the uni-reference dataset Segerstolpe et al.

```{r, message=F}
#fadista_ens_prop <- getPropbycond(ens_h = F_ensemble_h2, ens_d = F_ensemble_d2, w1h = 0.41, w2h = 0.59, w1d = 0.48, w2d = 0.52, ref = "ENSEMBLE")
#fadista_demo <- fadista_77@phenoData@data[,c("age","gender","hba1c","bmi")]
#fadista_ens_all <- cbind(fadista_ens_prop, fadista_demo[rownames(fadista_ens_prop),])

fadista_seger_prop <- getPropbycond(ens_h = F_ensemble_h2, ens_d = F_ensemble_d2, w1h = 0, w2h = 1, w1d = 0, w2d = 1, ref = "Segerstolpe")
fadista_baron_prop <- getPropbycond(ens_h = F_ensemble_h2, ens_d = F_ensemble_d2, w1h = 1, w2h = 0, w1d = 1, w2d = 0, ref = "Baron")
fadista_seger_all <- cbind(fadista_seger_prop, fadista_demo[rownames(fadista_seger_prop),])
fadista_baron_all <- cbind(fadista_baron_prop, fadista_demo[rownames(fadista_baron_prop),])

prop_lm <- function(propest){
  getlmtable <- function(ct){
    propest$ct <- propest[,ct]
    lm1 <- lm(ct ~ hba1c + age + bmi + gender, data = propest)
    s1 <- summary(lm1)
    sdt <- cbind(round(s1$coefficients,4), celltype = rep(ct,5))
    return(sdt)
  } 
  
  dtlm <- NULL
  for (ct in c("alpha","beta","delta","gamma","acinar","ductal")){
    dtlm0 <- getlmtable(ct)
    dtlm <- rbind(dtlm, dtlm0)
  }
  t <- as.data.frame(dtlm)
  t$param <- rownames(t)
  return(t)
}

segerproplm <- prop_lm(fadista_seger_all)
baronproplm <- prop_lm(fadista_baron_all)
ensproplm <- prop_lm(fadista_ens_all)

factor_to_num <- function(f){as.numeric(levels(f))[f]}
beta_pvalue <- rbind(factor_to_num(baronproplm[baronproplm$param=="hba1c",c(4)]),
                     factor_to_num(segerproplm[segerproplm$param =="hba1c",c(4)]),
                     factor_to_num(ensproplm[ensproplm$param=="hba1c",c(4)]))
colnames(beta_pvalue) <- ensproplm[ensproplm$param=="hba1c",5]
rownames(beta_pvalue) <- c("Baron","Segerstolpe","ENSEMBLE")

beta_pvalue
```


### Three-cell-line mixture

(1) Deconvolution of the real bulk sample: hard to distinguish the tumor cell types.

(2) Tree-guided deconvolution of real bulk sample. The choice of limit for LogFoldChange `LFC.lim=5` could be changed by users. A more robust way is to select a reasonable range of threshold, for example in this case, we choose from 3 to 6, and then estimate the proportions using different threshold. Say we calculated proportions from 3 to 6 by 0.1, we will derive 30 different estimated proportions results. We can calculate the mean from these results; or we can take the median of value of each estimated cell type, and normalize the sample-wise proportions to 1. Empirically, a good threshold value would cut the number of genes to 1000, which we believe is an efficient number to detect the inter-cell-type variations.

```{r, message=F, fig.height = 4, fig.width = 4, fig.align = "center"}
###(1)
sc3cl.basis <- SCDC_basis_ONE(qc.3cl$sc.eset.qc, ct.varname = "md_cluster", sample ="orig.ident")
df.d <- dist(t(log(sc3cl.basis$basis.mvw + 1e-10)), method = "euclidean")
hc1 <- hclust(df.d, method = "complete")
plot(hc1, cex = 0.8, hang = -1, main = "log(basis matrix)")

sc3cl2 <- sc.3cl
sc3cl2$md_cluster2 <- sc3cl2$md_cluster
sc3cl2$md_cluster2[!sc3cl2$md_cluster2 %in% "NormalFibroblast"] <- "Tumor"
# deconv_3cl <- SCDC_prop_ONE(bulk.eset = bulk3cl, sc.eset = sc3cl2, ct.varname = "md_cluster", sample = "orig.ident", ct.sub = unique(sc3cl$md_cluster), weight.basis = T)

###(2)
SCDC_3cl_tree <- SCDC_prop_ONE_subcl_marker(bulk.eset = bulk3cl, sc.eset = sc3cl2, ct.varname = "md_cluster",
                                             sample = "orig.ident", ct.sub = unique(sc3cl2$md_cluster),
                                             ct.fl.sub = unique(sc3cl2$md_cluster2) , weight.basis = T,
                                             fl.varname = "md_cluster2", select.marker = T, LFC.lim = 5)
SCDC_3cl_tree$prop.est
```


### Mammary Gland data

Cell types of interest are possibly related. We explore this by looking at the Pearson correlation between basis-vector for all cell types. We pool the single cells from two mouse mammary gland datasets Tabula Muris and SCDC by MNN. We construct basis matrix based on the batch-corrected single cell data, and investigate the inter-cell-type relationships.

```{r, message=F, fig.height = 4, fig.width = 4, fig.align = "center"}
### CELL-TYPE CORRELATION
mbmnn <- readRDS("H:/994/SCDC/data/S0206_mousebasismnn.rds")
print(mbmnn)
```

We deconvolve and ENSEMBLE for the 10X and FF samples separately.
```{r, message=F}
### the tree-guided deconvolution results:
#10x
tmouse.subcl <- readRDS("H:/994/SCDC/data/S0204_subcltmouse_3.rds")
perou.subcl <- readRDS("H:/994/SCDC/data/S0204_Perou10xsubcl3.rds")
#ff
perouff.subcl <- readRDS("H:/994/SCDC/data/S0204_subscperou_3ff.rds")
tmouseff.subcl<- readRDS("H:/994/SCDC/data/S0204_subcltmouse_3ff.rds")
ct.sub = c("endothelial","fibroblast","immune","luminal","basal")
#ensemble:
ens_subcl_perou10x <- SCDC_ENSEMBLE_subcl(bulk.eset = bulk10x, prop.list = list(tmouse.subcl = tmouse.subcl, perou.subcl = perou.subcl), 
                                          ct.sub = c("endothelial","fibroblast","immune","luminal","basal"), search.length = 0.01)

ens_subcl_perouff <- SCDC_ENSEMBLE_subcl(bulk.eset = bulkff, prop.list = list(tmouseff.subcl = tmouseff.subcl, perouff.subcl = perouff.subcl), 
                                         ct.sub = c("endothelial","fibroblast","immune","luminal","basal"), search.length = 0.01)

#ensembl proportions
ens10xout <- ens_subcl_perou10x$w_table
ensffout <- ens_subcl_perouff$w_table
## the 4th line is the mAD result:
ens_prop <- ens_subcl_perou10x$prop.list[[1]]$prop.est*ens10xout[4,1] +
  ens_subcl_perou10x$prop.list[[2]]$prop.est[rownames(ens_subcl_perou10x$prop.list[[1]]$prop.est),colnames(ens_subcl_perou10x$prop.list[[1]]$prop.est)]*ens10xout[4,2]
ens_propff <- ens_subcl_perouff$prop.list[[1]]$prop.est*ensffout[4,1]+
  ens_subcl_perouff$prop.list[[2]]$prop.est[rownames(ens_subcl_perouff$prop.list[[1]]$prop.est),colnames(ens_subcl_perouff$prop.list[[1]]$prop.est)]*ensffout[4,2]
```

Visualization of ENSEMBLE results for 10X and Fresh-frozen bulk samples.

```{r, message=F}
# visualize the ENSEMBLE results
perou <- qc.perou$sc.eset.qc
perou_pseudo_all <- generateBulk_allcells(perou, ct.varname = 'md_cluster', sample = 'subj', ct.sub = c("endothelial","fibroblast","immune","luminal","basal"))
dtmeltt<- melt(perou_pseudo_all$truep[,c("endothelial","fibroblast","immune","luminal","basal")])
dtmeltt$ref <- as.factor("SingleCells")
colnames(dtmeltt)[1:2] <- c("X1","X2")

### calculate pearson corr for 10x results
getPearson <- function(dt1, dt2, ct, subj){
  cor(c(dt1[subj,ct]), c(dt2[subj,ct]))
}
p10x_t <- getPearson(perou_pseudo_all$truep, ens_subcl_perou10x$prop.list[[1]]$prop.est, 
                     ct = c("endothelial","fibroblast","luminal","basal","immune"), subj = c("FVB3","FVB4"))
p10x_p <- getPearson(perou_pseudo_all$truep, ens_subcl_perou10x$prop.list[[2]]$prop.est, 
                     ct = c("endothelial","fibroblast","luminal","basal","immune"), subj = c("FVB3","FVB4"))
p10x_e <- getPearson(perou_pseudo_all$truep, ens_prop, 
                     ct = c("endothelial","fibroblast","luminal","basal","immune"), subj = c("FVB3","FVB4"))

dat_text10x <- data.frame(label = c("",paste("R =", round(c(p10x_t, p10x_p, p10x_e),2))), 
                          ref = as.factor(c("SingleCells","Tabula Muris", "Perou", "ENSEMBLE")),
                          X1 = rep("FVB3",4),
                          value =rep(0.9, 4),
                          X2 = rep("endothelial",4))

### calculate pearson corr for ff results
pff_t <- getPearson(perou_pseudo_all$truep, ens_subcl_perouff$prop.list[[1]]$prop.est, 
                    ct = c("endothelial","fibroblast","luminal","basal","immune"), subj = c("FVB3","FVB4"))
pff_p <- getPearson(perou_pseudo_all$truep, ens_subcl_perouff$prop.list[[2]]$prop.est, 
                    ct = c("endothelial","fibroblast","luminal","basal","immune"), subj = c("FVB3","FVB4"))
pff_e <- getPearson(perou_pseudo_all$truep, ens_propff, 
                    ct = c("endothelial","fibroblast","luminal","basal","immune"), subj = c("FVB3","FVB4"))

dat_textff <- data.frame(label = c("",paste("R =", round(c(pff_t, pff_p, pff_e),2))), 
                         ref = as.factor(c("SingleCells","Tabula Muris", "Perou", "ENSEMBLE")),
                         X1 = rep("FVB3",4),
                         value =rep(0.9, 4),
                         X2 = rep("endothelial",4))
```


#### Fresh-Frozen samples:

```{r, message=F, fig.height = 4, fig.width = 5, fig.align = "center"}
## ff
dtmelttm <- melt(ens_subcl_perouff$prop.list[[1]]$prop.est)
dtmelttm$ref <- as.factor("Tabula Muris")
dtmeltp <- melt(ens_subcl_perouff$prop.list[[2]]$prop.est)
dtmeltp$ref <- as.factor("Perou")
dtmelte <- melt(ens_propff)
dtmelte$ref <- as.factor("ENSEMBLE")
dtall <- rbind.data.frame(dtmeltt,dtmelttm, dtmeltp, dtmelte)
dtall <- dtall[dtall$X2 %in% c("endothelial","fibroblast","immune","luminal","basal"),]

ggplot(dtall, aes(x=X1, y=value, fill = factor(X2, levels = c("endothelial","fibroblast","luminal","basal","immune")))) + 
  geom_bar(stat = "identity") + xlab("") + ylab("Percentage") +
  theme(axis.text.x = element_text(angle = 30, hjust = 1, size=10),
        axis.text.y = element_text(size = 10),
        text = element_text(size = 10),
        plot.title = element_text(size=10, face = "bold"),
        plot.margin=unit(c(1,1,-5,0), "mm"),
        legend.position="top",legend.title = element_blank(),
        legend.text = element_text(size=8),
        legend.box.spacing = unit(0, "mm"))+
  scale_fill_manual(values=ct1) +
  guides(fill = guide_legend(nrow = 2))+
  facet_grid(cols = vars(ref)) +
  geom_text(data = dat_textff, 
            mapping = aes(x=X1, y=value,label = label),
            hjust   = 0.2, vjust   = 0)


```

#### 10X bulk samples:

```{r, message=F, fig.height = 4, fig.width = 5, fig.align = "center"}
dtmelttm <- melt(ens_subcl_perou10x$prop.list[[1]]$prop.est)
dtmelttm$ref <- as.factor("Tabula Muris")
dtmeltp <- melt(ens_subcl_perou10x$prop.list[[2]]$prop.est)
dtmeltp$ref <- as.factor("Perou")
dtmelte <- melt(ens_prop)
dtmelte$ref <- as.factor("ENSEMBLE")
dtall <- rbind.data.frame(dtmeltt,dtmelttm, dtmeltp, dtmelte)
dtall <- dtall[dtall$X2 %in% c("endothelial","fibroblast","luminal","basal","immune"),]

ggplot(dtall, aes(x=X1, y=value, fill = factor(X2, levels = c("endothelial","fibroblast","luminal","basal","immune")))) + 
  geom_bar(stat = "identity") + xlab("") + ylab("Percentage") +
  theme(axis.text.x = element_text(angle = 30, hjust = 1, size=10),
        axis.text.y = element_text(size = 10),
        text = element_text(size = 10),
        plot.title = element_text(size=10, face = "bold"),
        plot.margin=unit(c(1,1,-5,0), "mm"),
        legend.position="top",legend.title = element_blank(),
        legend.text = element_text(size=8),
        legend.box.spacing = unit(0, "mm"))+
  scale_fill_manual(values=ct1) +
  guides(fill = guide_legend(nrow = 2))+
  facet_grid(cols = vars(ref)) +
  geom_text(data = dat_text10x, 
            mapping = aes(x=X1, y=value,label = label),
            hjust   = 0.2, vjust   = 0)
```

From the deconvolution results for 10X and FF samples above, we see that the ENSEMBLE cell-type proportions have the highest Pearson correlation with the single cell clustering result. 
To this aspect, the estimation accuracy is improved by the SCDC ENSEMBLE method.
