---
title: "SCDC: leverage multiple Single-Cell RNA-seq reference sets for bulk gene expression DeConvolution"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{SCDC}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

**SCDC** adopts an ENSEMBLE method to integrate deconvolution results
across methods and datasets, giving reference data that are more close to the
bulk RNA-seq data higher weights, implicitly addressing the batch-effect confounding
when multiple scRNA-seq reference sets are available.

## Installation

```r
if (!require("devtools")) {
  install.packages("devtools")
}
devtools::install_github("meichendong/SCDC")
```

## Example 

### Data Input

SCDC takes `ExpressionSet` objects with raw read counts as input. `ExpressionSet` object is based on the package _Biobase_, which mainly stores data matrix (genes by samples), featureData (information about genes), and phenoData (information about samples).
We also enable a quick construction of the ExpressionSet object by the function `getESET()`. Here's a toy example:

```{r, message=F}
library(SCDC)
counts <- matrix(sample(0:500, 25), nrow = 5)
colnames(counts) <- paste("cell",1:5, sep = "")
rownames(counts) <- paste("gene",1:5, sep = "")
counts
fdata <- rownames(counts)
pdata <- cbind(cellname = colnames(counts), subjects = paste("patient", rep(1,5)))
eset <- getESET(counts, fdata = fdata, pdata = pdata)
class(eset)
```

As an illustration of the **SCDC** framework, we analyze the pancreatic islet data as presented in our paper. Three public scRNA-seq datasets from Baron et al.(2017), Segerstolpe et al.(2016) and Xin et al.(2016) are used; a bulk RNA-seq dataset from Fadista et al.(2014) is used in the real data analysis. The processed data is available for download at.

### Descriptive information of scRNA-seq data
For each singl-cell dataset with raw counts, we can first explore the demographic information by the visualization function `DemoPlot`. Here, single cells from each subject are summarized by total counts, counts per cell type, and the overall composition. This gives us the general understanding of a dataset. Take scRNA-seq data of healthy subjects from Segerstolpe et al. as an example, we derive the following plot:

```r
ct1 <- c("mediumorchid1","mediumpurple1","lightskyblue","seagreen1","yellow","tan1","azure3")
seger <- readRDS("segerstolpe.rds")
DemoPlot(seger, cluster = "cluster", sample = "sample", select.ct = c("alpha","beta","delta","gamma","ductal","acinar"), Palette = ct1)
```
![scRNA-seq from Segerstolpe et al.](demoplot.png) 

To make SCDC robust to single-cell clustering, a quality control procedure is performed as a first step to remove cells with questionable cell-type assignments, as well as cells with low library preparation and sequencing quality. Specifically, each single cell is treated as a “bulk” sample and its cell-type composition can be derived by a first-pass run of SCDC. 
The threshold of keeping a single cell from an assigned cluster is chosen as 0.7 in this illustration example. This user-defined threshold should be selected carefully, to achieve the effect of quality control of single cells without obstructing the performance on the downstream analysis. An unproper threshold could potentially lead to filtering out a large proportion of cells from rare cell types. To be conservative, this can be chosen from (0.5,0.7).

```r
seger.qc <- my_qc(seger, ct.varname = "cluster", sample = "sample", scsetname = "Segerstolpe",
                  ct.sub = c("alpha","beta","delta","gamma","ductal","acinar"), qcthreshold = 0.7) 
seger.qc$heatfig
```
![scRNA-seq from Segerstolpe et al. Clustering QC](cqc.png) 
Similarly we can perform clustering QC for single cells from only one subject using function `SCDC_qc_ONE()`.

