<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SCDC: leverage multiple Single-Cell RNA-seq reference sets for bulk gene expression DeConvolution • SCDC</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="SCDC: leverage multiple Single-Cell RNA-seq reference sets for bulk gene expression DeConvolution">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SCDC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/SCDC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/web/SCDC_data.html">SCDC example data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/meichendong/SCDC">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>SCDC: leverage multiple Single-Cell RNA-seq reference sets for bulk gene expression DeConvolution</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/meichendong/SCDC/blob/master/vignettes/SCDC.Rmd"><code>vignettes/SCDC.Rmd</code></a></small>
      <div class="hidden name"><code>SCDC.Rmd</code></div>

    </div>

    
    
<p><strong>SCDC</strong> adopts an ENSEMBLE method to integrate deconvolution results across methods and datasets, giving reference data that are more close to the bulk RNA-seq data higher weights, implicitly addressing the batch-effect confounding when multiple scRNA-seq reference sets are available.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="st">"devtools"</span>)) {
  <span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"devtools"</span>)
}
devtools<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/devtools/man/reexports.html">install_github</a></span>(<span class="st">"meichendong/SCDC"</span>)</code></pre></div>
</div>
<div id="example" class="section level2">
<h2 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h2>
<div id="data-input" class="section level3">
<h3 class="hasAnchor">
<a href="#data-input" class="anchor"></a>Data Input</h3>
<p>SCDC takes <code>ExpressionSet</code> objects with raw read counts as input. <code>ExpressionSet</code> object is based on the package <em>Biobase</em>, which mainly stores data matrix (genes by samples), featureData (information about genes), and phenoData (information about samples). We also enable a quick construction of the ExpressionSet object by the function <code><a href="../reference/getESET.html">getESET()</a></code>. Here’s a toy example:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(SCDC)
counts &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/sample.html">sample</a></span>(<span class="dv">0</span><span class="op">:</span><span class="dv">500</span>, <span class="dv">25</span>), <span class="dt">nrow =</span> <span class="dv">5</span>)
<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(counts) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"cell"</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dt">sep =</span> <span class="st">""</span>)
<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(counts) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"gene"</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dt">sep =</span> <span class="st">""</span>)
counts
<span class="co">#&gt;       cell1 cell2 cell3 cell4 cell5</span>
<span class="co">#&gt; gene1   308   110   185   183   251</span>
<span class="co">#&gt; gene2   159   418    77   115   425</span>
<span class="co">#&gt; gene3    22   154   336   313   441</span>
<span class="co">#&gt; gene4   203   160   424   491   188</span>
<span class="co">#&gt; gene5   231   125   476   170   359</span>
fdata &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(counts)
pdata &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="dt">cellname =</span> <span class="kw"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(counts), <span class="dt">subjects =</span> <span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"patient"</span>, <span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="dv">1</span>,<span class="dv">5</span>)))
eset &lt;-<span class="st"> </span><span class="kw"><a href="../reference/getESET.html">getESET</a></span>(counts, <span class="dt">fdata =</span> fdata, <span class="dt">pdata =</span> pdata)
<span class="kw"><a href="https://rdrr.io/r/base/class.html">class</a></span>(eset)
<span class="co">#&gt; [1] "ExpressionSet"</span>
<span class="co">#&gt; attr(,"package")</span>
<span class="co">#&gt; [1] "Biobase"</span></code></pre></div>
<p>As an illustration of the <strong>SCDC</strong> framework, we analyze the pancreatic islet data as presented in our paper. Three public scRNA-seq datasets from Baron et al.(2017), Segerstolpe et al.(2016) and Xin et al.(2016) are used; a bulk RNA-seq dataset from Fadista et al.(2014) is used in the real data analysis. The processed data is available for download at.</p>
</div>
<div id="descriptive-information-of-scrna-seq-data" class="section level3">
<h3 class="hasAnchor">
<a href="#descriptive-information-of-scrna-seq-data" class="anchor"></a>Descriptive Information of scRNA-seq Data</h3>
<p>For each singl-cell dataset with raw counts, we can first explore the demographic information by the visualization function <code>DemoPlot</code>. Here, single cells from each subject are summarized by total counts, counts per cell type, and the overall composition. This gives us the general understanding of a dataset. Take scRNA-seq data of healthy subjects from Segerstolpe et al. as an example, we derive the following plot:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ct1 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"mediumorchid1"</span>,<span class="st">"mediumpurple1"</span>,<span class="st">"lightskyblue"</span>,<span class="st">"seagreen1"</span>,<span class="st">"yellow"</span>,<span class="st">"tan1"</span>,<span class="st">"azure3"</span>)
seger &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="st">"segerstolpe.rds"</span>)
<span class="kw"><a href="../reference/DemoPlot.html">DemoPlot</a></span>(seger, <span class="dt">cluster =</span> <span class="st">"cluster"</span>, <span class="dt">sample =</span> <span class="st">"sample"</span>, <span class="dt">select.ct =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span>,<span class="st">"ductal"</span>,<span class="st">"acinar"</span>), <span class="dt">Palette =</span> ct1)</code></pre></div>
<div class="figure">
<img src="demoplot.png" alt="scRNA-seq from Segerstolpe et al."><p class="caption">scRNA-seq from Segerstolpe et al.</p>
</div>
<p>To make SCDC robust to single-cell clustering, a quality control procedure is performed as a first step to remove cells with questionable cell-type assignments, as well as cells with low library preparation and sequencing quality. Specifically, each single cell is treated as a “bulk” sample and its cell-type composition can be derived by a first-pass run of SCDC. The threshold of keeping a single cell from an assigned cluster is chosen as 0.7 in this illustration example. This user-defined threshold should be selected carefully, to achieve the effect of quality control of single cells without obstructing the performance on the downstream analysis. An unproper threshold could potentially lead to filtering out a large proportion of cells from rare cell types. To be conservative, this can be chosen from (0.5,0.7).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">seger.qc &lt;-<span class="st"> </span><span class="kw"><a href="../reference/my_qc.html">my_qc</a></span>(seger, <span class="dt">ct.varname =</span> <span class="st">"cluster"</span>, <span class="dt">sample =</span> <span class="st">"sample"</span>, <span class="dt">scsetname =</span> <span class="st">"Segerstolpe"</span>,
                  <span class="dt">ct.sub =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span>,<span class="st">"ductal"</span>,<span class="st">"acinar"</span>), <span class="dt">qcthreshold =</span> <span class="fl">0.7</span>) 
seger.qc<span class="op">$</span>heatfig</code></pre></div>
<p><img src="cqc.png" alt="scRNA-seq from Segerstolpe et al. Clustering QC"> Similarly we can perform clustering QC for single cells from only one subject using function <code><a href="../reference/SCDC_qc_ONE.html">SCDC_qc_ONE()</a></code>.</p>
</div>
<div id="scdc-on-simulated-data" class="section level3">
<h3 class="hasAnchor">
<a href="#scdc-on-simulated-data" class="anchor"></a>SCDC on Simulated Data</h3>
<p>SCDC allows users to construct pseudo bulk samples in two ways:</p>
<ul>
<li><p><code><a href="../reference/generateBulk_allcells.html">generateBulk_allcells()</a></code> Sum up gene-wise raw read counts per subject, and the true cell-type proportion is consistent with the single cell clustering result.</p></li>
<li><p><code><a href="../reference/generateBulk_norep.html">generateBulk_norep()</a></code>: Randomly sample single cells from each of the cell types of interest without replacement, and then sum up raw read counts per subject. The cell-type proportions are different from single cell clustering result.</p></li>
</ul>
<p>Here, we use the <code>ExpressionSet</code> objects after the clustering-QC of each scRNA-seq dataset. Read in the processed objects downloaded from our webpage:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># ExpressionSet objects from the three different scRNA-seq resources</span>
qc.seger &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="st">"qc_segerstolpe.rds"</span>)
qc.baron &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="st">"qc_baron.rds"</span>)
qc.xin &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="st">"qc_xin.rds"</span>)</code></pre></div>
<p>Then create the <em>pseudo bulk</em> samples:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## (1) using all single cells
pseudo.seger &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generateBulk_allcells.html">generateBulk_allcells</a></span>(qc.seger<span class="op">$</span>sc.eset.qc, <span class="dt">ct.varname =</span> <span class="st">"cluster"</span>, <span class="dt">sample =</span> <span class="st">"sample"</span>, <span class="dt">ct.sub =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span>))

## (2) using random proportions of single cells
pseudo.seger.rand &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generateBulk_norep.html">generateBulk_norep</a></span>(qc.seger<span class="op">$</span>sc.eset.qc, <span class="dt">ct.varname =</span> <span class="st">"cluster"</span>, <span class="dt">sample =</span> <span class="st">"sample"</span>, <span class="dt">ct.sub =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span>), <span class="dt">nbulk =</span> <span class="dv">3</span>)

## we know the true relative proportions for the pseudo bulk samples:
<span class="kw"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(pseudo.seger<span class="op">$</span>truep,<span class="dv">3</span>)
<span class="kw"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(pseudo.seger.rand<span class="op">$</span>true_p,<span class="dv">3</span>)</code></pre></div>
<p>The constructed <em>pseudo bulk</em> object includes the true cell-type proportion matrix <code>pseudo.seger$truep</code> and the <code>ExpressionSet</code> object <code>pseudo.seger$pseudo_eset</code>.</p>
<p>To perform deconvolution using one reference scRNA-seq dataset, use the <code><a href="../reference/SCDC_prop.html">SCDC_prop()</a></code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bulkseger.scseger &lt;-<span class="st"> </span><span class="kw"><a href="../reference/SCDC_prop.html">SCDC_prop</a></span>(<span class="dt">bulk.eset =</span> pseudo.seger<span class="op">$</span>pseudo_eset, <span class="dt">sc.eset =</span> qc.seger<span class="op">$</span>sc.eset.qc, <span class="dt">ct.varname =</span> <span class="st">"cluster"</span>, <span class="dt">sample =</span> <span class="st">"sample"</span>, <span class="dt">ct.sub =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span>), <span class="dt">truep =</span> pseudo.seger<span class="op">$</span>truep)
bulkseger.scbaron &lt;-<span class="st"> </span><span class="kw"><a href="../reference/SCDC_prop.html">SCDC_prop</a></span>(<span class="dt">bulk.eset =</span> pseudo.seger<span class="op">$</span>pseudo_eset, <span class="dt">sc.eset =</span> qc.baron<span class="op">$</span>sc.eset.qc, <span class="dt">ct.varname =</span> <span class="st">"cluster"</span>, <span class="dt">sample =</span> <span class="st">"sample"</span>, <span class="dt">ct.sub =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span>), <span class="dt">truep =</span> pseudo.seger<span class="op">$</span>truep)
bulkseger.scxin &lt;-<span class="st"> </span><span class="kw"><a href="../reference/SCDC_prop.html">SCDC_prop</a></span>(<span class="dt">bulk.eset =</span> pseudo.seger<span class="op">$</span>pseudo_eset, <span class="dt">sc.eset =</span> qc.xin<span class="op">$</span>sc.eset.qc, <span class="dt">ct.varname =</span> <span class="st">"cluster"</span>, <span class="dt">sample =</span> <span class="st">"sample"</span>, <span class="dt">ct.sub =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span>), <span class="dt">truep =</span> pseudo.seger<span class="op">$</span>truep)

bulkseger.scseger<span class="op">$</span>peval<span class="op">$</span>evals.table
<span class="co">#&gt;         RMSD     mAD      R</span>
<span class="co">#&gt; SCDC 0.03352 0.02553 0.9912</span>
bulkseger.scbaron<span class="op">$</span>peval<span class="op">$</span>evals.table
<span class="co">#&gt;         RMSD     mAD     R</span>
<span class="co">#&gt; SCDC 0.08337 0.06499 0.954</span>
bulkseger.scxin<span class="op">$</span>peval<span class="op">$</span>evals.table
<span class="co">#&gt;         RMSD     mAD      R</span>
<span class="co">#&gt; SCDC 0.19058 0.15449 0.9331</span></code></pre></div>
<p>We can observe the discrepancy of performance using different datasets. It’s clear that using single cells from the reference dataset Segerstolpe et al., we derive the highest Pearson correlation, and the lowest RMSD and mAD.</p>
<p>Next, we will explore the ENSEMBLE results. Using the pseudo bulk samples constructed above, we try to deconvolve using three reference datasets: Baron et al., Segerstolpe et al. and Xin et al. The search length is a parameter that is related to the grid search method.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pancreas.sc &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">baron =</span> qc.baron<span class="op">$</span>sc.eset.qc,
                    <span class="dt">seger =</span> qc.seger<span class="op">$</span>sc.eset.qc,
                    <span class="dt">xin =</span> qc.xin<span class="op">$</span>sc.eset.qc)

PS_ensemble &lt;-<span class="st"> </span><span class="kw"><a href="../reference/SCDC_ENSEMBLE.html">SCDC_ENSEMBLE</a></span>(<span class="dt">bulk.eset =</span> pseudo.seger<span class="op">$</span>pseudo_eset, <span class="dt">sc.eset.list =</span> pancreas.sc, <span class="dt">ct.varname =</span> <span class="st">"cluster"</span>,
                             <span class="dt">sample =</span> <span class="st">"sample"</span>, <span class="dt">truep =</span> pseudo.seger<span class="op">$</span>truep[<span class="kw"><a href="https://rdrr.io/r/stats/complete.cases.html">complete.cases</a></span>(pseudo.seger<span class="op">$</span>truep),], <span class="dt">ct.sub =</span>  <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span>), <span class="dt">search.length =</span> <span class="fl">0.01</span>)
PS_ensemble<span class="op">$</span>w_table
<span class="co">#&gt;                       w1   w2   w3    RMSD     mAD Pearson spearman_Y</span>
<span class="co">#&gt; max.Pearson_prop    0.01 0.94 0.05 0.03599 0.02797  0.9918    0.91372</span>
<span class="co">#&gt; min.mAD_Y           0.00 1.00 0.00 0.03352 0.02553  0.9912    0.91346</span>
<span class="co">#&gt; min.RMSD_Y          0.00 1.00 0.00 0.03352 0.02553  0.9912    0.91346</span>
<span class="co">#&gt; max.Spearman_corr_Y 0.00 0.96 0.04 0.03521 0.02741  0.9917    0.91381</span>
<span class="co">#&gt; min.mAD_Y_LAD       0.00 1.00 0.00 0.03352 0.02553  0.9912    0.91346</span>
<span class="co">#&gt; min.RMSD_Y_NNLS     0.00 1.00 0.00 0.03352 0.02553  0.9912    0.91346</span>
<span class="co">#&gt;                        RMSD_Y     mAD_Y</span>
<span class="co">#&gt; max.Pearson_prop    0.0003513 2.785e-05</span>
<span class="co">#&gt; min.mAD_Y           0.0003458 2.756e-05</span>
<span class="co">#&gt; min.RMSD_Y          0.0003458 2.756e-05</span>
<span class="co">#&gt; max.Spearman_corr_Y 0.0003490 2.778e-05</span>
<span class="co">#&gt; min.mAD_Y_LAD       0.0003458 2.756e-05</span>
<span class="co">#&gt; min.RMSD_Y_NNLS     0.0003458 2.756e-05</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#example">Example</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Meichen Dong.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9100.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
